feat: pull out HeaderAuthentication

r1b opened the pull request:
Test?

taion commented on flask_resty/authentication.py:
```diff
@@ -62,3 +65,45 @@ class NoOpAuthentication:
+        if authorization is None:
+            if self.credentials_arg is not None:
+                return flask.request.args.get(self.credentials_arg)
+            raise ApiError(401, {"code": "invalid_authorization.missing"})
```
i think we just return None for “no credentials” here. the check for “credentials are required” is proper to authZ

taion commented on flask_resty/authentication.py:
```diff
@@ -62,3 +65,45 @@ class NoOpAuthentication:
+        return self.get_credentials_from_authorization(authorization)
+
+    def get_credentials_from_authorization(self, authorization):
+        if self.header_scheme is None:
```
could just default this to Bearer. the other schemes all have handshakes and challenges that were not implementing

r1b commented:
Ping, I renamed this back to `HeaderAuthentication` since it can be instantiated as-is with a default scheme. Added tests, not sure why coverage is complaining.

r1b commented:
Ah nvm it was because I broke JWT :P
